homeassistant/components/backup/config.py

```diff
@@ -2,56 +2,116 @@
 
 from __future__ import annotations
 
-from dataclasses import dataclass
+from dataclasses import dataclass, field, replace
 from datetime import datetime
-from typing import Self, TypedDict
+from typing import TYPE_CHECKING, Self, TypedDict
 
-from homeassistant.core import HomeAssistant
+from cronsim import CronSim
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.event import async_track_point_in_time
 from homeassistant.helpers.storage import Store
 from homeassistant.helpers.typing import UNDEFINED, UndefinedType
+from homeassistant.util import dt as dt_util
 
 from .const import DOMAIN
 
+if TYPE_CHECKING:
+    from .manager import BackupManager
+
+CRON_PATTERN_DAILY = "0 2 * * *"
+CRON_PATTERN_WEEKLY = "0 2 * * {}"
+DELAY_SAVE = 30
 STORAGE_KEY = DOMAIN
 STORAGE_VERSION = 1
 
 
 class StoredBackupConfig(TypedDict):
     """Represent the stored backup config."""
 
+    addons_included: list[str] | None
+    agent_ids: list[str]
+    database_included: bool
+    folders_included: list[str] | None
     last_automatic_backup: datetime | None
     max_copies: int | None
+    name: str | None
+    password: str | None
+    schedule: StoredBackupSchedule
+
+
+class StoredBackupSchedule(TypedDict):
+    """Represent the stored backup schedule."""
+
+    daily: bool
+    never: bool
+    weekday: str | None
```

Comment:
Would it make sense to use an enum?

Reply:
Do you mean a nine member enum (daily + never + all seven days)? The validation will be simpler. I'll try it out.

Reply:
Oh it was mostly that there are only 7 days, so I think it makes sense to have an enum with 7 values instead of a string, but I think that would also work

Reply:
Yeah, the days should definitely be an enum.

Reply:
Based on the complex logic below, it would be worth trying the 9-digit enum idea